### PR TITLE
Update and flesh out browserify

### PR DESCRIPTION
**Pull request by jaschaephraim**

### CODE REVIEW

**Comment by jvilk** [comment]
@jaschaephraim I just noticed this! I did the same thing:

https://github.com/DefinitelyTyped/DefinitelyTyped/pull/6833

We should join forces.

**Comment by jvilk** [comment]
I've updated my PR to support some of the typings I missed or had incorrect with respect to yours. Let me know if I overlooked anything!

File: browserify/browserify.d.ts [hunk 1]
```diff
@@ -1,41 +1,66 @@
-// Type definitions for Browserify
-// Project: http://browserify.org/
-// Definitions by: Andrew Gaspar <https://github.com/AndrewGaspar/>
```
Comment (vvakame) [inline browserify/browserify.d.ts]: please keep original author names. like this https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/jquery/jquery.d.ts#L3

**Comment by jaschaephraim** [comment]
@jvilk Looks like you got some updated definitions through. Nice work!

**Comment by jvilk** [comment]
Thanks! They would've been much worse, had I not checked them against yours -- you exposed a serious bug in mine.

**Pull request closed without merging**